### PR TITLE
Chore/controller balance transfer

### PR DESCRIPTION
**Pull request by b0xtch**
## Changelog

- Support requestBalance
- Support requestTransfer

### Type of changes included:

- [X] Components
- [X] Business Logic
- [ ] Bug fixes
- [X] Styling
- [ ] Code Refactor

### Screenshots/Gifs:

*Place you changes screenshots here*

### Notes:

This is dependant on the inpage provider located [here](https://github.com/Psychedelic/plug-inpage-provider/pull/7)

### Tested on:

- [X] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

File: source/Background/Controller.js [hunk 1]
```diff
@@ -143,5 +144,65 @@ backgroundController.exposeController(
...
+
+    extension.windows.create({
+      url,
+      type: 'popup',
+      width: TRANSFER_SIZES.width,
+      height: TRANSFER_SIZES.detailHeightSmall,
+      top: 65,
+      left: metadata.pageWidth - TRANSFER_SIZES.width,
+    });
+  },
```
Comment (rocky-fleek) [inline source/Background/Controller.js:191]: we are missing the connected check here

File: source/Background/Controller.js [hunk 2]
```diff
@@ -143,5 +144,65 @@ backgroundController.exposeController(
...
+backgroundController.exposeController(
+  'handleRequestTransfer',
+  async (opts, args, callId, portId) => {
+    const { callback } = opts;
+
+    const [transfer] = args;
+    const keyringHandler = getKeyringHandler(HANDLER_TYPES.SEND_ICP, keyring);
+    const transferResponse = await keyringHandler(transfer);
+
+    callback(null, true);
```
Comment (rocky-fleek) [inline source/Background/Controller.js:203]: is the true answering to the `requestTransfer` call? If so, we should check if the transaction succeeded or not first. Or even if it was accepted, like if the user rejected the trx this still goes through right? it should come in `transfer.state` and I think it can be `accepted` or `rejected`

File: webpack.config.js [hunk 3]
```diff
@@ -235,7 +230,7 @@ module.exports = {
   ],
 
   optimization: {
-    minimize: nodeEnv !== 'development',
+    minimize: false,
```
Comment (rocky-fleek) [inline webpack.config.js:233]: revert this change plz :pray:

**Review by rocky-fleek — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 3 above)
Comment (b0xtch) [reply webpack.config.js]: 5fa94c9

File: source/Background/Controller.js (hunk 1 above)
Comment (b0xtch) [reply source/Background/Controller.js]: `requestBalance` checks connection through keyringHandler
`requestTransfer` added in tbdf2a2b

File: source/Background/Controller.js [hunk 4]
```diff
@@ -16,15 +15,19 @@ const backgroundController = new BackgroundController({
     'plug-content-script',
     'notification-port',
     'app-connection-port',
```
Comment (ChriLnth) [inline source/Background/Controller.js:17]: we can remove this one

File: source/Background/Controller.js [hunk 5]
```diff
@@ -54,13 +57,18 @@ backgroundController.exposeController(
...
     const { message, sender } = opts;
 
-    storage.set({
-      [domainUrl]: {
-        url: domainUrl,
-        name,
-        status: CONNECTION_STATUS.pending,
-        icon,
-      },
+    storage.get('apps', (response) => {
```
Comment (rocky-fleek) [inline source/Background/Controller.js:60]: shouldnt this be a set???

File: source/Pages/Notification/components/AppConnection/index.jsx [hunk 6]
```diff
@@ -13,16 +11,18 @@ import i18n from 'i18next';
 
 import { CONNECTION_STATUS } from '@shared/constants/connectionStatus';
 import { Provider } from 'react-redux';
-import store from '../../redux/store';
-import initConfig from '../../locales';
+import store from '../../../../redux/store';
+import initConfig from '../../../../locales';
 import useStyles from './styles';
+import SIZES from '../Transfer/constants';
+import { Layout } from '../../../../components';
```
Comment (rocky-fleek) [inline source/Pages/Notification/components/AppConnection/index.jsx:19]: alises

File: source/Pages/Notification/components/Balance/index.jsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+const portRPC = new PortRPC({
+  name: 'transfer-port',
+  target: 'bg-script',
+  timeout: 20000,
+});
+
+portRPC.start();
+
+const Balance = ({
+  args: accountId, metadata, callId, portId,
```
Comment (rocky-fleek) [inline source/Pages/Notification/components/Balance/index.jsx:14]: not accountId, its subaccount

File: source/Pages/Notification/components/Transfer/hooks/useRequests.jsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  data.push(...[
+    {
+      label: t('common.fee'),
+      component: <DataDisplay value={<AssetFormat value={requests[currentRequest]?.args?.fee || DEFAULT_FEE} asset={CURRENCIES.get('ICP').value} />} />,
+    },
+    {
+      label: t('common.memo'),
+      component: <DataDisplay value={requests[currentRequest]?.args?.memo || t('common.null')} />,
+    },
+  ]);
```
Comment (rocky-fleek) [inline source/Pages/Notification/components/Transfer/hooks/useRequests.jsx:103]: we should refactor this

File: source/hooks/useApps.jsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      if (lsApps) {
+        setApps(lsApps);
+      }
+    });
+  }, []);
+
+  useEffect(() => {
+    storage.set({
+      apps,
+    });
```
Comment (rocky-fleek) [inline source/hooks/useApps.jsx:34]: I'd add `apps` as a dependency

**Review by rocky-fleek — APPROVED** [review]
After all of these changes, LGTM
